### PR TITLE
Add support for Patch Ops add/remove

### PR DESCRIPTION
**Pull request by schuylermartin45**
- Adds support for `add`/`remove` in `RecipeParser.patch()`
- Adds unit testing
- `add` supports `append`, as per the RFC with the `-` character
- Fixes the quote bug described in PKG-2982

### CODE REVIEW

File: percy/tests/test_recipe_parser.py [hunk 1]
```diff
@@ -415,13 +408,115 @@ def test_patch_schema_validation() -> None:
...
+                {
+                    "op": "test",
+                    "path": "",
+                    "value": 42,
+                }
+            )
+        )
+
+    # add
+    assert not (
```
Comment (JeanChristopheMorinPerso) [inline percy/tests/test_recipe_parser.py:432]: I just realized this now (and I'm a little lazy to check myself): When a patch fails to be applied, how could a user know what went wrong?

File: percy/tests/test_recipe_parser.py [hunk 2]
```diff
@@ -545,6 +641,175 @@ def test_patch_test() -> None:
...
+            "value": {
+                "George": ["How'd you like your own TV show?", "You're on"],
+                "Stanley": ["Ok"],
+            },
+        }
+    )
+
+    # Sanity check: validate all modifications
+    assert parser.is_modified()
+    assert parser.render() == load_file(f"{TEST_FILES_PATH}/simple-recipe_test_patch_add.yaml")
```
Comment (JeanChristopheMorinPerso) [inline percy/tests/test_recipe_parser.py:747]: I like that you do a render at the end to compare and make sure the output is as expected. Should we do `get` operations after each patch too to confirm that the patches are applied correctly individually?

File: percy/tests/test_recipe_parser.py [hunk 3]
```diff
@@ -545,6 +641,175 @@ def test_patch_test() -> None:
     assert not parser.is_modified()
 
 
+def test_patch_add() -> None:
+    """
+    Tests the `remove` patch op.
```
Comment (JeanChristopheMorinPerso) [inline percy/tests/test_recipe_parser.py:646]:
```suggestion
    Tests the `add` patch op.
```

**Review by JeanChristopheMorinPerso — APPROVED** [review]

File: percy/tests/test_recipe_parser.py (hunk 1 above)
Comment (schuylermartin45) [reply percy/tests/test_recipe_parser.py:432]: `patch()` returns with a boolean value, indicating success or failure to apply a patch.

I use a `bool` over an exception because the `test` op returns a `bool` and I decided that if one op needed to return a `bool`, all should return a `bool`.

The RFC is a little vague about error handling other than if a failure occurs, the failure should not modify the file. That should be the case here, unless there are bugs or missing edge cases.

File: percy/tests/test_recipe_parser.py (hunk 2 above)
Comment (schuylermartin45) [reply percy/tests/test_recipe_parser.py:747]: I do something like that somewhere else in the unit tests, I can't remember where off the top of my head.

It's not a bad idea, I just don't know if it's overkill. I purposefully try not to re-use path names to prevent side-effects of previous commands.
